### PR TITLE
Imminent fixes for bugs found on master branch

### PR DESCRIPTION
**Pull request by arthurtilly**
* The vanilla level checks define for Yoshi is inverted, causing him to require 120 stars when it is off and appear at 0 stars when it's on

* The downwarp fix results in Mario levitating in midair when grabbing Bowser midair. While downwarps should still be fixed, the change should be reverted immediately until a better fix is made.

* Some checks of Mario's floor class were using the wrong defines, which can lead to unexpected behavior in the event anyone wants to reorder surface types. By default the SURFACE_CLASS_SLIPPERY and SURFACE_SLIPPERY defines have the same value, which is why this mistake is hard to notice.

* The firsty frames define was implemented poorly, not allowing for vanilla firsty behavior no matter what the values were set to. This has been reverted, while avoiding the UB in the original vanilla code.

* Removed the ledge grab changes that fix QSLGs and change the false ledgegrab define since Arceveti wanted to in his PR

### CODE REVIEW

**Review by gheskett — APPROVED** [review]
The downwarping doesn't break anything else unique to HackerSM64, does it?

**Comment by arthurtilly** [comment]
cant be 100% sure, i very highly doubt it, what i do know is that not downwarping breaks something much more important lol

**Review by Reonu — APPROVED** [review]
I agree with all the changes, including the reverts. The reverted code can still be merged back to the repo if a bug-free version is made

File: src/engine/surface_collision.c [hunk 1]
```diff
@@ -704,7 +704,7 @@ s32 find_water_level(s32 x, s32 z) { // TODO: Allow y pos
 
             // If the location is within a water box and it is a water box.
             // Water is less than 50 val only, while above is gas and such.
-            if (loX < x && x < hiX && loZ < z && z < hiZ && val < 50) {
+            if (loX <= x && x <= hiX && loZ <= z && z <= hiZ && val < 50) {
```
Comment (gheskett) [inline src/engine/surface_collision.c:707]: What does this do exactly? Vanilla doesn't use `<=`.

Comment (arthurtilly) [reply src/engine/surface_collision.c:707]: in vanilla one of these is a float, for some reason in hacker it was changed to both ints. this results in a unit-sized gap between adjacent waterboxes (most notably the waterboxes in JRB) which causes mario to upwarp occasionally when travelling between them. in vanilla this was the size of a floating point imprecision so it basically never happened but now it's much easier to trigger.

Comment (gheskett) [reply src/engine/surface_collision.c:707]: In that case I'd assume it technically makes more sense to leave `hiX` and `hiZ` as <, but it probably doesn't really matter anyway if we leave it like this.

**Review by gheskett — APPROVED** [review]

File: include/config/config_movement.h [hunk 2]
```diff
@@ -68,14 +68,11 @@
...
-// The last frame after hitting a wall that will be considered a firsty when wallkicking.
-#define FIRSTY_LAST_FRAME 1
-
 // The maximum angle the player can wall kick, in degrees. 0..90. To allow 45 degree wall kicks, you must supply `46` to allow 45 and under.
 #define WALL_KICK_DEGREES 45
 
-// This is vanilla behavior, disable it to allow ledge grabbing regardless of floor pitch.
-// #define LEDGE_GRABS_CHECK_SLOPE_ANGLE
+// Makes Mario unable to ledge grab steep slopes to prevent false ledge grabs.
+#define DONT_LEDGE_GRAB_STEEP_SLOPES
```
Comment (thecozies) [inline include/config/config_movement.h:75]: this should be positively worded ideally, I don't have a suggestion atm

Comment (gheskett) [reply include/config/config_movement.h:75]: `LEDGE_GRAB_STEEP_SLOPES`? :)

File: src/game/mario_actions_automatic.c [hunk 3]
```diff
@@ -565,11 +565,9 @@ s32 act_ledge_grab(struct MarioState *m) {
     if (m->actionTimer < 10) {
         m->actionTimer++;
     }
-#ifdef LEDGE_GRABS_CHECK_SLOPE_ANGLE
```
Comment (thecozies) [inline src/game/mario_actions_automatic.c:568]: todo here, this causes slope stutters which sucks. ig this should check if the floor is slippery too

File: src/game/mario_step.c [hunk 4]
```diff
@@ -404,8 +403,10 @@ struct Surface *check_ledge_grab(struct MarioState *m, struct Surface *prevWall,
     if (ledgeFloor == NULL
         || (*ledgeFloor) == NULL
         || ledgePos[1] < nextPos[1] + 100.0f
+#ifdef DONT_LEDGE_GRAB_STEEP_SLOPES
         || (*ledgeFloor)->normal.y < COS25 // H64 TODO: check if floor is actually slippery
```
Comment (thecozies) [inline src/game/mario_step.c:407]: re: checking if floor is actually slippery

File: src/game/mario_actions_automatic.c (hunk 3 above)
Comment (arthurtilly) [reply src/game/mario_actions_automatic.c:568]: well regardless this is the exact same check as the one i replaced it with so it wont really behave too differently to how it was already, still a direct upgrade

File: README.md [hunk 5]
```diff
@@ -14,7 +14,7 @@ This is a fork of the ultrasm64 repo by CrashOveride which includes the followin
 - **ArcticJaguar725**: Most audio configuration and layout changes, colored ia4 text, floombas, various bugfixes, and more
 - **CowQuack**: Adjustable skybox sizes, area-specific skybox function
 - **thecozies**: Water surface types, general maintenance, and time
-- **MrComit**: General use object defines, JUMP_KICK_FIX, LEDGE_GRABS_CHECK_SLOPE_ANGLE
```
Comment (thecozies) [inline README.md:17]: wait you renamed this but completely removed the credits from comit lmao

Comment (gheskett) [reply README.md:17]: Comit freeloading smh

**Pull request merged**